projects/packages/my-jetpack/_inc/state/actions.js

```diff
@@ -109,6 +111,9 @@ function requestProductStatus( productId, data, { select, dispatch } ) {
 			.then( freshProduct => {
 				dispatch( setIsFetchingProduct( productId, false ) );
 				dispatch( setProduct( freshProduct ) );
+				if ( 'POST' === method ) {
+					globalDispatch( CONNECTION_STORE_ID ).refreshConnectedPlugins();
```

Comment:
shouldn't try to refresh the plugins list when the product is deleted, too? Actually isn't clear when/why we need to refresh the plugins list, right? At least here.

I know that, under the hood, some plugins are indirectly activated when activating a product, but maybe refreshing always the plugins list won't hurt.

Being said in another way, is not possible that the active plugins list would change when the product is deactivated?

Reply:
Yes, the list could be changed on deactivation, but since we don't have the deactivation button anymore I didn't add it there...

It wouldn't hurt to refresh always, but it's an unnecessary request. 

What do you think? Should I add the DELETE method to the list or just update every time?

Reply:
or simply remove the `if ( 'POST' === method ) {` line?

> It wouldn't hurt to refresh always, but it's an unnecessary request.

It's true. That's why I mentioned that isn't clear when the plugin list should be refreshed, because also the list won't change when activating any product, right?
I mean, I think it's going to perform a request unnecessarily. What are the products that can change the active plugins list?

Reply:
> What are the products that can change the active plugins list?

All of them, depending on the combination of plugins you have installed.

If I have only Backup and activate VideoPress jetpack will be installed. Also CRM and Boost.

If I have jetpack installed, Backup and Scan will not change the list, bu CRM and Boost will...

But I think it's fine to the request everytime. I'll remove the check

Reply:
https://github.com/Automattic/jetpack/pull/23358/commits/f887dd1c372b8ff56e894c732834b4070c0ceac0